sklearn/utils/estimator_checks.py

```diff
@@ -2080,7 +2081,18 @@ def check_classifiers_train(
         X_b -= X_b.min()
 
     if readonly_memmap:
-        X_m, y_m, X_b, y_b = create_memmap_backed_data([X_m, y_m, X_b, y_b])
+        # OpenBLAS is known to segfault with unaligned data on the Prescott architecture
+        # See: https://github.com/scipy/scipy/issues/14886
+        has_prescott_openblas = any(
+            True
+            for info in threadpool_info()
+            if info["internal_api"] == "openblas"
+            and info.get("architecture").lower() == "prescott"
```

Comment:
For old versions of threadpoolctl, this will raise `AttributeError` on `None`. Better use aligned data when we cannot introspect the architecture.

```suggestion
            and info.get("architecture", "prescott").lower() == "prescott"
```